src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/action/delete/TransportDeleteTransformsAction.kt

```diff
@@ -0,0 +1,90 @@
+package com.amazon.opendistroforelasticsearch.indexmanagement.transform.action.delete
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.IndexManagementPlugin.Companion.INDEX_MANAGEMENT_INDEX
+import org.elasticsearch.ElasticsearchStatusException
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.bulk.BulkRequest
+import org.elasticsearch.action.bulk.BulkResponse
+import org.elasticsearch.action.delete.DeleteRequest
+import org.elasticsearch.action.get.MultiGetRequest
+import org.elasticsearch.action.get.MultiGetResponse
+import org.elasticsearch.action.support.ActionFilters
+import org.elasticsearch.action.support.HandledTransportAction
+import org.elasticsearch.client.Client
+import org.elasticsearch.common.inject.Inject
+import org.elasticsearch.index.IndexNotFoundException
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.tasks.Task
+import org.elasticsearch.transport.TransportService
+
+class TransportDeleteTransformsAction @Inject constructor(
+    transportService: TransportService,
+    val client: Client,
+    actionFilters: ActionFilters
+) : HandledTransportAction<DeleteTransformsRequest, BulkResponse>(
+    DeleteTransformsAction.NAME, transportService, actionFilters, ::DeleteTransformsRequest
+) {
+
+    override fun doExecute(task: Task, request: DeleteTransformsRequest, actionListener: ActionListener<BulkResponse>) {
+
+        // TODO: if metadata id exists delete the metadata doc else just delete transform
+
+        // Use Multi-Get Request
+        var getRequest = MultiGetRequest()
+        request.ids.forEach { id ->
+            getRequest.add(MultiGetRequest.Item(INDEX_MANAGEMENT_INDEX, id))
+        }
+
+        client.multiGet(getRequest, object : ActionListener<MultiGetResponse> {
+            override fun onResponse(response: MultiGetResponse) {
+                try {
+                    bulkDelete(response, request.ids, actionListener)
+                } catch (e: Exception) {
+                    actionListener.onFailure(e)
+                }
+            }
+
+            override fun onFailure(e: Exception) = actionListener.onFailure(e)
+        })
+    }
+
+    private fun bulkDelete(response: MultiGetResponse, ids: List<String>, actionListener: ActionListener<BulkResponse>) {
+        var enabledIDs = mutableListOf<String>()
+        response.responses.forEach {
+            // TODO: Check if the source is actually transform document
+            if (it.response != null && it.response.isExists) {
+                val source = it.response.source
```

Comment:
Can you add a check that the document is indeed transform type before including this in delete? and if the response is failed